viam-cartographer_test.go

```diff
@@ -1,392 +1,216 @@
+			DataDirectory: dataDir,
+			DataRateMsec:  dataRateMsec,
 			UseLiveData:   &_true,
 		}
 
-		_, err := createSLAMService(t, attrCfg, logger, false, false, testExecutableName)
+		_, err := testhelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
 		test.That(t, err, test.ShouldBeError,
-			errors.New("configuring camera error: error getting camera gibberish for slam service: \"gibberish\" missing from dependencies"))
+			errors.New("configuring lidar camera error: error getting lidar camera "+
+				"gibberish for slam service: \"gibberish\" missing from dependencies"))
 	})
 
-	closeOutSLAMService(t, name)
-}
-
-func TestCartographerNew(t *testing.T) {
-	logger := golog.NewTestLogger(t)
-	name, err := slamTesthelper.CreateTempFolderArchitecture(logger)
-	test.That(t, err, test.ShouldBeNil)
-
-	t.Run("New cartographer service with good lidar in slam mode 2d", func(t *testing.T) {
-		grpcServer, port := setupTestGRPCServer(t)
+	t.Run("Successful creation of cartographer slam service with good lidar", func(t *testing.T) {
+		grpcServer, port := testhelper.SetupTestGRPCServer(t, logger)
 		attrCfg := &slamConfig.AttrConfig{
 			Sensors:       []string{"good_lidar"},
 			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: name,
-			DataRateMsec:  validDataRateMS,
+			DataDirectory: dataDir,
+			DataRateMsec:  dataRateMsec,
 			Port:          "localhost:" + strconv.Itoa(port),
 			UseLiveData:   &_true,
 		}
 
-		svc, err := createSLAMService(t, attrCfg, logger, false, true, testExecutableName)
+		svc, err := testhelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
 		test.That(t, err, test.ShouldBeNil)
 
 		grpcServer.Stop()
 		test.That(t, utils.TryClose(context.Background(), svc), test.ShouldBeNil)
 	})
 
-	t.Run("New cartographer service with lidar that errors during call to NextPointCloud", func(t *testing.T) {
-		attrCfg := &slamConfig.AttrConfig{
-			Sensors:       []string{"bad_lidar"},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: name,
-			DataRateMsec:  validDataRateMS,
-			UseLiveData:   &_true,
-		}
-
-		_, err = createSLAMService(t, attrCfg, logger, false, false, testExecutableName)
-		test.That(t, err, test.ShouldBeError,
-			errors.Errorf("runtime slam service error: error getting data from sensor: %v", attrCfg.Sensors[0]))
-	})
-
-	t.Run("New cartographer service with camera without NextPointCloud implementation", func(t *testing.T) {
+	t.Run("Failed creation of cartographer slam service with invalid sensor "+
+		"that errors during call to NextPointCloud", func(t *testing.T) {
 		attrCfg := &slamConfig.AttrConfig{
 			Sensors:       []string{"invalid_sensor"},
 			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: name,
-			DataRateMsec:  validDataRateMS,
+			DataDirectory: dataDir,
+			DataRateMsec:  dataRateMsec,
 			UseLiveData:   &_true,
 		}
 
-		_, err = createSLAMService(t, attrCfg, logger, false, false, testExecutableName)
```

Comment:
[q] Some tests like this one passed `success = false` to createSLAMService, but now CreateSLAMService always does what used to be `success = true`. I couldn't quickly tell why this change was ok - do you mind explaining?

Reply:
I'll gladly explain. Using `success` in the `createSLAMService` had no purpose. It simply ignored the value of `svc` in two test cases (for orbslam) when logically we expect the test to fail, but `success` is passed in as `true` (which makes no sense in itself). `svc` is `nil` in those two tests anyways - which is what I expect in my version of `createSLAMService`. 

Basically - when `err != nil` then `svc == nil` and vice versa. `success` simply ignored the value of `svc` in two test cases, when logically it didn't make sense to ignore the value of `svc`, cause we should expect it to be `nil`, and it is in fact `nil`.

Happy to discuss in person if it helps.

@zaporter-work this is also going to be relevant to you when you refactor orbslam. You can also safely remove `success` despite those two test cases.

Reply:
Ah, I see! Thanks for explaining, and sounds good to me.